spec/unit/application/client_spec.rb

```diff
@@ -420,6 +420,7 @@
   before do
     @original_argv = ARGV.dup
     ARGV.clear
+    allow(::File).to receive(:read).with("/proc/sys/crypto/fips_enabled").and_call_original
```

Comment:
makes me a bit twitchy.  should really be `allow(::File).to recieve(:read).and_call_original` (which is much clearer that the problem is that you're working around the stub on the next line blocking other arguments).

and it feels like it should a bit more explicitly test this.  setup the non-fips case and explicitly mock it out to run non-fips even on a fips box, and then mock it to fips and make sure the behavior is correct even on a non-fips box.